datasets/web_nlg/web_nlg.py

```diff
@@ -238,8 +252,15 @@ def _generate_examples(self, filedirs):
         id_ = 0
         for xml_location in filedirs:
             for xml_file in sorted(glob(pjoin(xml_location, "*.xml"))):
-                test_cat = xml_file.split("/")[-1][:-4] if "webnlg_challenge_2017/test" in xml_file else ""
+                if (
+                    "webnlg_challenge_2017/test" in xml_file
+                    or "release_v3.0/en/test" in xml_file
+                    or "release_v3.0/ru/test" in xml_file
+                ):
+                    test_cat = xml_file.split("/")[-1][:-4]
```

Comment:
Unfortunately this kind of check may not work on Windows. Indeed Windows uses backslashes and not simple slashes between directory names.

So I would change it this way:
```suggestion
                # windows may use backslashes so we first need to replace them with slashes
                xml_file_path_with_slashes = "/".join(Path(xml_file).parts)
                if (
                    "webnlg_challenge_2017/test" in xml_file_path_with_slashes
                    or "release_v3.0/en/test" in xml_file_path_with_slashes
                    or "release_v3.0/ru/test" in xml_file_path_with_slashes
                ):
                    test_cat = xml_file_path_with_slashes.split("/")[-1][:-4]
```
Let me know what you think